test/components/DayPicker_spec.jsx

```diff
@@ -790,11 +791,14 @@ describe('DayPicker', () => {
 
     describe('#componentDidUpdate', () => {
       let updateStateAfterMonthTransitionSpy;
+      let setCalendarMonthWeeksSpy;
+
       beforeEach(() => {
         updateStateAfterMonthTransitionSpy = sinon.stub(
           DayPicker.prototype,
           'updateStateAfterMonthTransition',
         );
+        setCalendarMonthWeeksSpy = sinon.stub(PureDayPicker.prototype, 'setCalendarMonthWeeks');
```

Comment:
If you unskip the containing describe, do your new tests pass?

Reply:
Ah! They don't 👎 In fact, none of the tests in that section are passing (I didn't even notice the skip). It looks like it's been skipped for about two years now.

I can take it on and go back and fix all those tests. Should I fix mine for this PR and make a separate PR for fixing the other tests? Or lump it all into this one?

Reply:
I don't mind if your new test happens to be skipped; but it's important that the new test *would* be passing, so that when we unskip this section, we don't have to go fix it :-)

Reply:
The issue here is that _all_ the lifecycle tests are being skipped because  `componentDidUpdate` isn't being called without fully mounting the component. Full mount isn't working right now and is throwing: `Error: It looks like you called `mount()` without a global document being loaded.`

I'm happy to implement something like `jsdom-global` to allow mounting to work with the setup but I'm wondering if that should be a separate PR from this one? Or if I should take another approach to testing these methods? I could obviously stub it out but that seems intuitive to me as that wouldn't _really_ be mimicking production.

Reply:
If you could enable these tests in another PR, we could merge that prior to this one - that’d be great!

Reply:
#1415 addresses this - @ljharb